roundabout-core/src/test/java/at/fhv/itm3/s2/roundabout/mocks/RouteGenerator.java

```diff
@@ -85,100 +87,48 @@ private void initializeRouteWithTwoStreetSections() {
-
-        // initialize source and route
-        AbstractSource source3 = new RoundaboutSourceMock(model, "", false, street3_1, 2, this, RouteType.ONE_STREETSECTION_ONE_ONEWAYSTREET);
-
-        IRoute route3 = new Route();
-        route3.addSource(source3);
-        route3.addSection(street3_1);
-        route3.addSection(street3_2);
-        route3.addSection(roundaboutSink3);
-
-        routes.put(2, route3);
-
-
-        // INITIALIZE ROUTE WITH TWO ONEWAYSTREETS
-        // initialize streets and sink
-        RoundaboutSink roundaboutSink4 = new RoundaboutSink(model, "", false);
-        OneWayStreet oneWayStreet4_2 = new OneWayStreet(model, "", false, 2, roundaboutSink4, 10);
-        Street street4_2 = new OneWayStreetAdapter(oneWayStreet4_2, model, "", false);
-        OneWayStreet oneWayStreet4_1 = new OneWayStreet(model, "", false, 2, street4_2, 10);
-        Street street4_1 = new OneWayStreetAdapter(oneWayStreet4_1, model, "", false);
+    private void initializeRouteWithIntersection() {
+        float turnaroundTime = 60;
+        float[] phaseShiftTimes = new float[]{0.0F, 10.0F, 20.0F};
+        double intersectionTraverseTime = 5.0;
+        double accelerationTime = 2.0;
+        double yellowDuration = turnaroundTime / 8;
+        double greenDuration = turnaroundTime / 2 - yellowDuration;
+        int outDirection = 1;
+        int inDirection = 0;
+
+        // initialize intersection
+        RoundaboutIntersection intersection = new RoundaboutIntersection(model, "", false, 2);
+        intersection.setServiceDelay(accelerationTime);
+        FixedCirculationController ic = ModelFactory.getInstance(model).createOneWayController(
+                intersection,
+                greenDuration,
+                yellowDuration,
+                phaseShiftTimes[0]
+        );
+        intersection.attachController(ic);
+
+        // initialize streets
+        AbstractProSumer street1 = new StreetSection(10.0, model, "", false);
+        AbstractProSumer street2 = new StreetSection(10.0, model, "", false);
+
+        // initialize sink
+        RoundaboutSink sink = new RoundaboutSink(model, "", false);
+
+        // initialize source
+        AbstractSource source = new RoundaboutSourceMock(model, "", false, (StreetSection)street1, 2, this, RouteType.STREETSECTION_INTERSECTION_STREETSECTION);
+
+        // connect streets with intersection
+        IntersectionController.getInstance().setIntersectionInDirectionMapping(intersection, street1, inDirection);
+        IntersectionController.getInstance().setIntersectionOutDirectionMapping(intersection, street2, outDirection);
+        intersection.attachProducer(inDirection, street1.toProducer());
+        intersection.attachConsumer(outDirection, street2.toConsumer());
+        intersection.createConnectionQueue(street1.toProducer(), new AbstractConsumer[]{street2.toConsumer()}, new double[]{intersectionTraverseTime}, new double[]{1.0});
 
```

Comment:
complicated process to connect the streets with the intersection
would it be possible to create the connection queue somehow automatically?

Reply:
please create a issue, this is behaviour from another branch